_drafts/2015-02-05-play2-californium-raspi.md

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: Integrating Californium with the Raspberry Pi
+author: lucas
+---
+
+On our Raspberry Pi web interface we want to display the data from all our plant nodes in "realtime" (in double-quotes because, well, we're talking about websites). In this post we will be introducing our technology stack that will be running on the Raspberry Pi and explain step by step how all the piece fit and can be made to work together. 
+
+
+<!-- more -->
+
+![](images/play2-californium/architecture.png)
+
+The graphic shows an abstract overview of our setup. On the bottom left we have one of our plant nodes which communicates with the [Eclipse Californium](https://www.eclipse.org/californium/) Server via the CoAP protocol, both of which have been introduced {% postlink 2015-02-18-what-is-coap here %}. To develop the web interface we have chosen the [Play framework](https://playframework.com/) because it runs inside the JVM and thus integrates well with Californium. The communication between the Californium Server and the web application is established through an [Akka](http://akka.io/) Actor, which is the conventional solution for asynchronous communication within Play. Lastly, the user's browser communicates with the Play application server using HTTP as well as the WebSocket protocol for "realtime" updates. Communication between actors is highlighted in red. No protocol is mentioned because no serialization of the messages is necessary, since the subsystems operate inside the same Java Virtual Machine, thus making it possible to simply pass Java objects around.
```

Comment:
suggestion: add a line break after “which have been introduced {% postlink 2015-02-18-what-is-coap here %}.” to show that youre switching topics.